homeassistant/components/notify/html5.py

```diff
@@ -169,15 +169,32 @@ def post(self, request):
             return self.json_message(
                 humanize_error(data, ex), HTTP_BAD_REQUEST)
 
-        name = ensure_unique_string('unnamed device', self.registrations)
+        name = self.find_registration_name(data)
+        previous_registration = self.registrations.get(name)
 
         self.registrations[name] = data
 
-        if not save_json(self.json_path, self.registrations):
+        try:
+            save_json(self.json_path, self.registrations)
```

Comment:
I know it's not your code but we shouldn't do I/O inside the event loop.

Rewrite this to be:

```python
hass = request.app['hass']

yield from hass.async_add_job(save_json, self.json_path, self.registrations)
```

Reply:
Done. I updated `delete` as well with that code. Fortunately the tests were not too hard to update :)